src/Sulu/Bundle/CoreBundle/Resources/config/list_builder.xml

```diff
@@ -0,0 +1,72 @@
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="sulu_core.list_builder.field_descriptor_factory"
+                 class="Sulu\Component\Rest\ListBuilder\Metadata\FieldDescriptorFactory">
+            <argument type="service" id="sulu_core.list_builder.metadata.provider.chain"/>
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.provider.chain"
+                 class="Sulu\Component\Rest\ListBuilder\Metadata\Provider\ChainProvider">
+            <argument type="collection">
+                <argument type="service" id="sulu_core.list_builder.metadata.provider.general"/>
+                <argument type="service" id="sulu_core.list_builder.metadata.provider.doctrine"/>
+            </argument>
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.file_locator" class="Metadata\Driver\FileLocator">
+            <argument type="collection"/>
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.provider.general"
+                 class="Sulu\Component\Rest\ListBuilder\Metadata\Provider\MetadataProvider">
+            <argument type="service" id="sulu_core.list_builder.metadata.provider.general.factory"/>
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.provider.general.factory" class="Metadata\MetadataFactory">
+            <argument type="service" id="sulu_core.list_builder.metadata.provider.general.driver.xml"/>
+            <argument>Metadata\ClassHierarchyMetadata</argument>
+            <argument>%kernel.debug%</argument>
+            <!-- TODO
+            <call method="setCache">
+                <argument type="service" id="???" />
+            </call>
+            -->
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.provider.general.driver.xml"
+                 class="Sulu\Component\Rest\ListBuilder\Metadata\General\Driver\XmlDriver">
+            <argument type="service" id="sulu_core.list_builder.metadata.file_locator"/>
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.provider.doctrine"
+                 class="Sulu\Component\Rest\ListBuilder\Metadata\Provider\MetadataProvider">
+            <argument type="service" id="sulu_core.list_builder.metadata.provider.doctrine.factory"/>
+        </service>
+
+        <service id="sulu_core.list_builder.metadata.provider.doctrine.factory" class="Metadata\MetadataFactory">
+            <argument type="service" id="sulu_core.list_builder.metadata.provider.doctrine.driver.xml"/>
+            <argument>Metadata\ClassHierarchyMetadata</argument>
+            <argument>%kernel.debug%</argument>
+            <!-- TODO
+            <call method="setCache">
+                <argument type="service" id="???" />
+            </call>
```

Comment:
See the MassiveSearchBundle for an example of a cache implementation: https://github.com/massiveart/MassiveSearchBundle/blob/develop/Resources/config/metadata.xml#L24